src/main/java/com/karl/Server.java

```diff
@@ -0,0 +1,55 @@
+package com.karl;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class Server {
+  private ServerSocket serverSocket;
+  private Socket clientSocket;
+  private PrintWriter out;
+  private BufferedReader in;
+
+  public Server(ServerSocket serverSocket) {
+    this.serverSocket = serverSocket;
+  }
+
+  public void start() throws IOException {
+    clientSocket = accept();
+    in = createRequestFrom(clientSocket);
+    String inputLine;
+    while ((inputLine = in.readLine()) != null) {
```

Comment:
For future refactoring, can we DI PrintWriter and BufferedReader?

Reply:
I moved this stuff into the socket wrapper class so that it is tied to socket wrapper, and not to the server itself. Let me know if you still think that I should do this, and how best. As arguments to the socket wrapper class?

Reply:
ok great! If you find that the tests are easier to write and assert then this redesign is valuable. If you feel that it is still not, perhaps DI the reader and writer in the SocketWrapper constructor is another avenue to try. DI has the benefit of letting you mock the collaborator and assert when things get called in the mock. DI in the constructor is one option, you have used DI using method injection which is another option :) 
This is all just for future consideration and refactoring as you build out the design - not a blocker.